diff --git a/servant-lucid.cabal b/servant-lucid.cabal
index acbc7dc..6f4d851 100644
--- a/servant-lucid.cabal
+++ b/servant-lucid.cabal
@@ -38,7 +38,7 @@ library
                      , http-media >=0.6.4   && <0.9
                      , lucid      >=2.9.8   && <2.10
                      , text       >=1.2.3.0 && <1.3
-                     , servant    >=0.17    && <0.18
+                     , servant    >=0.17    && <0.19
 
   if !impl(ghc >= 8.0)
     build-depends:     semigroups >=0.18.4  && <0.20
@@ -57,7 +57,7 @@ test-suite example
       base
     , lucid
     , servant-lucid
-    , servant-server >=0.14     && <0.18
+    , servant-server >=0.14     && <0.19
     , wai            >=3.0.3.0  && <3.3
     , warp           >=3.0.13.1 && <3.4
   default-language: Haskell2010
