diff --git a/stylish-haskell.cabal b/stylish-haskell.cabal
index 8e9dffd..0cd7654 100644
--- a/stylish-haskell.cabal
+++ b/stylish-haskell.cabal
@@ -55,7 +55,7 @@ Library
     aeson            >= 0.6    && < 1.5,
     base             >= 4.8    && < 5,
     bytestring       >= 0.9    && < 0.11,
-    Cabal            >= 2.4    && < 3.1,
+    Cabal            >= 2.4    && < 3.3,
     containers       >= 0.3    && < 0.7,
     directory        >= 1.2.3  && < 1.4,
     filepath         >= 1.1    && < 1.5,
@@ -81,7 +81,7 @@ Executable stylish-haskell
     aeson            >= 0.6    && < 1.5,
     base             >= 4.8    && < 5,
     bytestring       >= 0.9    && < 0.11,
-    Cabal            >= 2.4    && < 3.1,
+    Cabal            >= 2.4    && < 3.3,
     containers       >= 0.3    && < 0.7,
     directory        >= 1.2.3  && < 1.4,
     filepath         >= 1.1    && < 1.5,
@@ -141,7 +141,7 @@ Test-suite stylish-haskell-tests
     aeson            >= 0.6    && < 1.5,
     base             >= 4.8    && < 5,
     bytestring       >= 0.9    && < 0.11,
-    Cabal            >= 2.4    && < 3.1,
+    Cabal            >= 2.4    && < 3.3,
     containers       >= 0.3    && < 0.7,
     directory        >= 1.2.3  && < 1.4,
     filepath         >= 1.1    && < 1.5,
