diff --git a/text-trie.cabal b/text-trie.cabal
index 65c1f92..7af741e 100644
--- a/text-trie.cabal
+++ b/text-trie.cabal
@@ -73,9 +73,9 @@ Library
     -- The lower bounds are more restrictive than necessary.
     -- But then, we don't maintain any CI tests for older
     -- versions, so these are the lowest bounds we've verified.
-    Build-Depends: base       >= 4.5   && < 4.13
+    Build-Depends: base       >= 4.5   && < 4.15
                  , text       >= 1.2.3 && < 1.2.4
-                 , binary     >= 0.5.1 && < 0.8.7
+                 , binary     >= 0.5.1 && < 0.8.9
 
 ------------------------------------------------------------
 Test-Suite test-text-trie
@@ -90,10 +90,10 @@ Test-Suite test-text-trie
                   , FromListBench.Text.Encode
                   , TrieFile.Text
     build-depends: text-trie
-                 , base       >= 4.5   && < 4.13
+                 , base       >= 4.5   && < 4.15
                  , bytestring >= 0.9.2 && < 0.11
                  , text       >= 1.2.3 && < 1.2.4
-                 , binary     >= 0.5.1 && < 0.8.7
+                 , binary     >= 0.5.1 && < 0.8.9
                  , QuickCheck
                  , HUnit
                  , smallcheck
