diff --git a/src/Data/Solidity/Prim/Tuple.hs b/src/Data/Solidity/Prim/Tuple.hs
index 5ac5798..cd516a8 100644
--- a/src/Data/Solidity/Prim/Tuple.hs
+++ b/src/Data/Solidity/Prim/Tuple.hs
@@ -2,6 +2,8 @@
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE StandaloneDeriving  #-}
 {-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleContexts    #-}
 
 -- |
 -- Module      :  Data.Solidity.Prim.Tuple
@@ -26,7 +28,6 @@ import           Data.Solidity.Abi           (AbiGet, AbiPut, AbiType (..))
 import           Data.Solidity.Abi.Generic   ()
 import           Data.Solidity.Prim.Tuple.TH (tupleDecs)
 
-deriving instance GHC.Generic (OneTuple a)
 instance Generic (OneTuple a)
 
 instance AbiType a => AbiType (OneTuple a) where
diff --git a/src/Language/Solidity/Abi.hs b/src/Language/Solidity/Abi.hs
index 607d361..e094802 100644
--- a/src/Language/Solidity/Abi.hs
+++ b/src/Language/Solidity/Abi.hs
@@ -144,6 +144,18 @@ instance Ord Declaration where
     compare DFallback {} DFunction {} = GT
     compare DFallback {} DEvent {} = GT
 
+$(deriveToJSON
+   (defaultOptions {
+       sumEncoding = TaggedObject "type" "contents"
+       , constructorTagModifier = over _head toLower . drop 1
+       , fieldLabelModifier = over _head toLower . drop 3 })
+   ''Declaration)
+
+$(deriveJSON (defaultOptions {
+    sumEncoding = TaggedObject "stateMutability" "contents"
+  , constructorTagModifier = fmap toLower . drop 2 })
+    ''StateMutability)
+
 instance FromJSON Declaration where
   parseJSON = withObject "Declaration" $ \o -> do
     t :: Text <- o .: "type"
@@ -159,18 +171,6 @@ instance FromJSON Declaration where
             Nothing -> o .: "constant"
             Just sm -> pure $ sm `elem` [SMPure, SMView]
 
-$(deriveToJSON
-   (defaultOptions {
-       sumEncoding = TaggedObject "type" "contents"
-       , constructorTagModifier = over _head toLower . drop 1
-       , fieldLabelModifier = over _head toLower . drop 3 })
-   ''Declaration)
-
-$(deriveJSON (defaultOptions {
-    sumEncoding = TaggedObject "stateMutability" "contents"
-  , constructorTagModifier = fmap toLower . drop 2 })
-    ''StateMutability)
-
 
 -- | Contract Abi is a list of method / event declarations
 newtype ContractAbi = ContractAbi { unAbi :: [Declaration] }
diff --git a/web3-solidity.cabal b/web3-solidity.cabal
index 2f5449a..dddb988 100644
--- a/web3-solidity.cabal
+++ b/web3-solidity.cabal
@@ -46,22 +46,22 @@ library
       src
   ghc-options: -funbox-strict-fields -Wduplicate-exports -Whi-shadowing -Widentities -Woverlapping-patterns -Wpartial-type-signatures -Wunrecognised-pragmas -Wtyped-holes -Wincomplete-patterns -Wincomplete-uni-patterns -Wmissing-fields -Wmissing-methods -Wmissing-exported-signatures -Wmissing-monadfail-instances -Wmissing-signatures -Wname-shadowing -Wunused-binds -Wunused-top-binds -Wunused-local-binds -Wunused-pattern-binds -Wunused-imports -Wunused-matches -Wunused-foralls -Wtabs
   build-depends:
-      OneTuple >0.2 && <0.3
-    , aeson >1.2 && <1.6
-    , base >4.11 && <4.15
-    , basement >0.0 && <0.1
-    , bytestring >0.10 && <0.11
-    , cereal >0.5 && <0.6
-    , data-default >0.7 && <0.8
-    , generics-sop >0.3 && <0.6
-    , memory >0.14 && <0.16
-    , memory-hexstring ==1.0.*
-    , microlens >0.4 && <0.5
-    , parsec >3.1 && <3.2
-    , tagged >0.8 && <0.9
-    , template-haskell >2.11 && <2.17
-    , text >1.2 && <1.3
-    , web3-crypto ==1.0.*
+      OneTuple
+    , aeson
+    , base
+    , basement
+    , bytestring
+    , cereal
+    , data-default
+    , generics-sop
+    , memory
+    , memory-hexstring
+    , microlens
+    , parsec
+    , tagged
+    , template-haskell
+    , text
+    , web3-crypto
   default-language: Haskell2010
 
 test-suite tests
@@ -94,24 +94,24 @@ test-suite tests
       src
   ghc-options: -funbox-strict-fields -Wduplicate-exports -Whi-shadowing -Widentities -Woverlapping-patterns -Wpartial-type-signatures -Wunrecognised-pragmas -Wtyped-holes -Wincomplete-patterns -Wincomplete-uni-patterns -Wmissing-fields -Wmissing-methods -Wmissing-exported-signatures -Wmissing-monadfail-instances -Wmissing-signatures -Wname-shadowing -Wunused-binds -Wunused-top-binds -Wunused-local-binds -Wunused-pattern-binds -Wunused-imports -Wunused-matches -Wunused-foralls -Wtabs -threaded -rtsopts -with-rtsopts=-N
   build-depends:
-      OneTuple >0.2 && <0.3
-    , aeson >1.2 && <1.6
-    , base >4.11 && <4.15
-    , basement >0.0 && <0.1
-    , bytestring >0.10 && <0.11
-    , cereal >0.5 && <0.6
-    , data-default >0.7 && <0.8
-    , generics-sop >0.3 && <0.6
-    , hspec >=2.4.4 && <2.8
-    , hspec-contrib >=0.4.0 && <0.6
-    , hspec-discover >=2.4.4 && <2.8
-    , hspec-expectations >=0.8.2 && <0.9
-    , memory >0.14 && <0.16
-    , memory-hexstring ==1.0.*
-    , microlens >0.4 && <0.5
-    , parsec >3.1 && <3.2
-    , tagged >0.8 && <0.9
-    , template-haskell >2.11 && <2.17
-    , text >1.2 && <1.3
-    , web3-crypto ==1.0.*
+      OneTuple
+    , aeson
+    , base
+    , basement
+    , bytestring
+    , cereal
+    , data-default
+    , generics-sop
+    , hspec
+    , hspec-contrib
+    , hspec-discover
+    , hspec-expectations
+    , memory
+    , memory-hexstring
+    , microlens
+    , parsec
+    , tagged
+    , template-haskell
+    , text
+    , web3-crypto
   default-language: Haskell2010
